lib/gtype.ml

```diff
@@ -420,3 +420,9 @@ let normalise_global_t (global_t : global_t) =
   Map.fold
     ~init:(Map.empty (module ProtocolName))
     ~f:normalise_protocol global_t
+
+let global_t_of_ast (ast : Syntax.scr_module) : global_t =
```

Comment:
@becharrens I'm not sure how this differs from `global_t_of_module`, could you maybe explain the difference?

Reply:
I think that the main difference is that this function not only converts the scr_module into a global_t, but it also normalises the protocols (unfolding and flattening choices) and replaces recursive constructs with nested protocols and nested protocol calls. `global_t_of_module` simply converts the ast into a global type and builds the protocol mapping. I think it may be a bit confusing after coming back to it now...

Reply:
Filed #121 to resolve it later